### ISSUE TITLE
_clamp increases bound values when using scale()

### ISSUE THREAD

**Issue by gopeter** [craftyjs/Crafty#954]
When `_clamp` is called and `scale()` is set, the bounds increases by every call of `_clamp` because the values are multiplicated by its reference. That's how `console.log(bound.max.x)` looks like when the player moves:

![bildschirmfoto 2015-08-17 um 10 56 07](https://cloud.githubusercontent.com/assets/1242960/9301509/4cac72da-44cf-11e5-89bd-b69d1342b5c8.png)

Making a deep copy of the bound object solves the problem. Because I don't know how to make a deep copy correctly, I can't provide a pull request. I fixed it very dirty:

```
var b = this.bounds || Crafty.map.boundaries();
var bound = JSON.parse(JSON.stringify(b));

bound.max.x *= this._scale;
bound.min.x *= this._scale;
bound.max.y *= this._scale;
bound.min.y *= this._scale;
```

Just to let you know :)

Another problem I've encountered is the calculation of the the viewport when using `follow()` and `scale()` together. The camera doesn't stop on the bottom and right border when reaching the bound because the `scale` factor isn't included in the calculation, so I fixed it like this:

`if (Crafty.viewport.y < -bound.max.y + Crafty.viewport.height)` becomes `if (Crafty.viewport.y < (-bound.max.y + Crafty.viewport.height) / this._scale)` (and so on).

You can check out a demo here: https://debd.com/clients/oetinger/kk-game-pirates-viewport-pirates/test.html

If you are going to the very bottom you'll see that the camera does't stop correctly and always shows the player in the middle of the screen even though the game border/bound is already reached.

**Comment by gopeter** [comment]
Oh, `Crafty.clone()` does the job! :smiley: 

I'll make a pull request.

**Comment by mpetrovic** [comment]
shouldn't bounds increase with scale, since now the absolute playing area
has changed?

**Comment by gopeter** [comment]
The bounds increases every time the player moves. Bounds should increase only once per scale (I updated my issue, sorry for confusing).

**Comment by starwed** [comment]
Fixed in #955
